websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,279 @@
+const presence = new Presence({
+		clientId: "970743721404530798"
+	}),
+	browsingTimestamp = Math.floor(Date.now());
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, hostname } = window.location;
+	switch (hostname) {
+		case "sonicretro.org": {
+			presenceData.smallImageKey = "home_512";
+			presenceData.smallImageText = "sonicretro.org";
+			if (search) {
+				presenceData.largeImageKey = "search";
+				if (search === "?s=")
+					presenceData.details = "Searching for something...";
```

Comment:
did you come across a situation where the search query was empty? How did that happen?

Reply:
> did you come across a situation where the search query was empty? How did that happen?

Website got a weird bug where u can just press enter when focused on the search bar, that's when the search comes back empty. Just there in case it happens.

Reply:
in that case they're not searching for anything

Reply:
> in that case they're not searching for anything

It still triggers the search page, so just to save the code from errors ig. Can it stay?

Reply:
Sure, but use URLSearchParams instead because there may be other query options

Reply:
this wasn't addressed

Reply:
> this wasn't addressed

There's nothing to address here, there's NEVER any instance of other query options on this page.

Reply:
That can change at any time and the user can also add those yourself. You are **not** allowed to resolve conversations without addressing them for the record

Reply:
changed it